### ISSUE TITLE
A way to limit video length/A way to edit start and end times for the video (i.g. 01:22-12:30)

### ISSUE THREAD

**Issue by MESAresearch** [ytb2mp3/youtube-mp3-downloader#57]
I'm trying to prevent users from being able to upload videos that have like a 10 hour long length so as to not bog up my servers. I can't seem to find anything about this on the docs or when searching online. Is this not a feature yet? 

Also being able to edit start and end times to clip the video would be nice. I of course know that's possible with ffmpeg but if it could be done more efficiently at the beginning so that actually only the clipped portion of the video is captured/downloaded by the YouTube-mp3-downloader package. Thanks

**Comment by fdciabdul** [comment]
you can read the documentary from ytdl

**Comment by kevin4dhd** [comment]
any solution?

**Comment by ytb2mp3** [comment]
I'm open to contributions / PRs, feel free to add this functionality

**Comment by fdciabdul** [comment]
i thought u need this one , but u need to install ytdl globally with npm i -g ytdl
```javascript
ytdl.getInfo("ID").then(info => {
if (info.length_seconds > 3000){
console.log("file is too big");
}else{
///

}
```
> any solution?

**Comment by kevin4dhd** [comment]
My code for 10 minutes

```
var YoutubeMp3Downloader = require("youtube-mp3-downloader");
var ffmpeg = require('@ffmpeg-installer/ffmpeg');
var ytdl = require('ytdl-core');

//Configure YoutubeMp3Downloader with your settings
var YD = new YoutubeMp3Downloader({
    "ffmpegPath": ffmpeg.path, // FFmpeg binary location
    "outputPath": "./youtube_mp3/mp3", // Output file location (default: the home directory)
    "youtubeVideoQuality": "highestaudio", // Desired video quality (default: highestaudio)
    "queueParallelism": 2, // Download parallelism (default: 1)
    "progressTimeout": 2000, // Interval in ms for the progress reports (default: 1000)
    "allowWebm": false // Enable download from WebM sources (default: false)
});

let url = "https://www.youtube.com/watch?v=RwJbRV_SSAk&t=35s";
url = url.replace('https://www.youtube.com/watch?v=', '');
url = url.replace('https://youtu.be/', '');
url = url.substring(0, url.lastIndexOf('&'));

try {
    ytdl.getInfo(url).then(info => {
        console.log(info);
        if (info.lengthSeconds > 600) {
            console.log("file is too big");
        } else {
            console.log("e.e");
        }
    });
} catch (e) {
    console.log("La URL no existe");
}
```

**Comment by kevin4dhd** [comment]
thanks @fdciabdul

**Comment by fdciabdul** [comment]
> My code for 10 minutes
> 
> ```
> var YoutubeMp3Downloader = require("youtube-mp3-downloader");
> var ffmpeg = require('@ffmpeg-installer/ffmpeg');
> var ytdl = require('ytdl-core');
> 
> //Configure YoutubeMp3Downloader with your settings
> var YD = new YoutubeMp3Downloader({
>     "ffmpegPath": ffmpeg.path, // FFmpeg binary location
>     "outputPath": "./youtube_mp3/mp3", // Output file location (default: the home directory)
>     "youtubeVideoQuality": "highestaudio", // Desired video quality (default: highestaudio)
>     "queueParallelism": 2, // Download parallelism (default: 1)
>     "progressTimeout": 2000, // Interval in ms for the progress reports (default: 1000)
>     "allowWebm": false // Enable download from WebM sources (default: false)
> });
> 
> let url = "https://www.youtube.com/watch?v=RwJbRV_SSAk&t=35s";
> url = url.replace('https://www.youtube.com/watch?v=', '');
> url = url.replace('https://youtu.be/', '');
> url = url.substring(0, url.lastIndexOf('&'));
> 
> try {
>     ytdl.getInfo(url).then(info => {
>         console.log(info);
>         if (info.lengthSeconds > 600) {
>             console.log("file is too big");
>         } else {
>             console.log("e.e");
>         }
>     });
> } catch (e) {
>     console.log("La URL no existe");
> }
> ```

for getting video id , you just set the regex like this 
```javascript
var videoid = url.match(/(?:https?:\/{2})?(?:w{3}\.)?youtu(?:be)?\.(?:com|be)(?:\/watch\?v=|\/)([^\s&]+)/);

console.log(videoid[1]);
```

**Comment by MESAresearch** [comment]
Idk what happened but for some reason now it only converts the first 10 seconds of videos into audio?? Is anyone else getting this problem

**Comment by fdciabdul** [comment]
> Idk what happened but for some reason now it only converts the first 10 seconds of videos into audio?? Is anyone else getting this problem

nope , send your coding on here

**Comment by MESAresearch** [comment]
> > Idk what happened but for some reason now it only converts the first 10 seconds of videos into audio?? Is anyone else getting this problem
> 
> nope , send your coding on here

Oh wait nvm. Think it was just a problem with my server I guess because it works now

**Comment by MESAresearch** [comment]
Yeah wait actually. I still am getting the 10 second problem. But for me it only occurs for certain videos. Is this happening to anyone else? https://www.youtube.com/watch?v=wuO4_P_8p-Q

**Comment by MESAresearch** [comment]
> Yeah wait actually. I still am getting the 10 second problem. But for me it only occurs for certain videos. Is this happening to anyone else? https://www.youtube.com/watch?v=wuO4_P_8p-Q

function youtube_parser(url){
              var regExp = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/;
              var match = url.match(regExp);
              return (match&&match[7].length==11)? match[7] : false;
            }

            var isTooLong;

            ytdl.getInfo(youtube_parser(ytAudioURL)).then(info => { //checking yt video length 
              if (info.videoDetails.lengthSeconds > 600){
                console.log('cannot convert videos longer than 10min to audio');
                isTooLong = true;
              }



              if(!isTooLong){
                var YD = new YoutubeMp3Downloader({
                  "ffmpegPath": "../../../usr/bin/ffmpeg",        // FFmpeg binary location
                  "outputPath": `./public/uploads/${folder}`,    
                  "youtubeVideoQuality": "highestaudio",  
                  "queueParallelism": 2,                  
                  "progressTimeout": 2000,                
                  "allowWebm": false                      
                });
                //Download video and save as MP3 file
                YD.download(youtube_parser(ytAudioURL));
                
                YD.on("finished", function(err, data) {
                    var ytAudFileName = data.videoTitle + '.mp3';
                    postModel.findOneAndUpdate({postNumber:counted},{mp3File:ytAudFileName},function(err,doc){
                      if (err) {
                        throw err;
                      }
                      console.log('youtube mp3 file '+ytAudFileName+' has been saved to folder: '+folder);
                      console.log(doc);
                    });
            
                });
              } //YT download is in callback so as to wait for the verdict on if isTooLong is true or not


            });



            here's my code
